File: 0104-maximum-depth-of-binary-tree/solution.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    int getDepth(TreeNode node, int depth){
        int now_depth = depth + 1;
        if(node.left != null){
            now_depth = Math.Max(now_depth,getDepth(node.left,depth+1));
        }
        if(node.right != null){
            now_depth = Math.Max(now_depth,getDepth(node.right,depth+1));
        }
        return now_depth;
    }
    public int MaxDepth(TreeNode root) {
        int ans = 0;
        if(root!=null)ans=getDepth(root,0);
        return ans;
    }
}

File: 0104-maximum-depth-of-binary-tree/solution.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
    int getDepth(TreeNode* node){
        if(node==nullptr)return 0;
        int dpt=max(getDepth(node->left),getDepth(node->right));
        dpt++;
        return dpt;
    }
public:
    int maxDepth(TreeNode* root) {
        return getDepth(root);
    }
};
